package/tvheadend-app-osmc/files/DEBIAN/postinst

```diff
@@ -41,6 +94,61 @@ if [ "$1" = "configure" ]; then
 		fi
 	fi
 
-	if ischroot; then exit 0; fi
+	# start tvh so that it updates any old user files
+	echo Starting tvheadend to finalise setup for osmc
 	if systemctl is-enabled tvheadend.service >/dev/null; then systemctl start tvheadend.service; fi
+	#systemctl restart tvheadend
+	sleep 5
+
+	# add failed privilege and htsp to all users
+	function addparameter(){
+		if [ $(sed -n /\"$1\":/,/\]/{/$2/p} "$USERDIR/$f" | wc -l) -eq 0 ]; then
+			if [ $(sed -nr /\"$1\":/,/\]/{/[a-z]/p} "$USERDIR/$f" | wc -l) -gt 1 ]; then
+				sed -i /\"$1\":/a\\\"$2\", "$USERDIR/$f"
+			else
+				sed -i /\"$1\":/a\\\"$2\" "$USERDIR/$f"
+			fi
+#			echo $2 added to $1 in "$USERDIR/$f"
+#		else
+#			echo $2 already in $1 in "$USERDIR/$f"
+		fi
+	}
+
+	USERDIR=/home/osmc/.hts/tvheadend/accesscontrol
+	USERS=($(ls $USERDIR))
+	for f in ${USERS[@]}; do
+		for f in ${USERS[@]}; do
+			addparameter dvr failed 
+			for p in streaming dvr; do
+				addparameter $p htsp 
+			done
+		done
+	done
+
+	# osmc must have a password if tvh is not to ignore it
+	if [ -n $OSMCUSERFILE ] && ! grep "\"username\": \"osmc\"" /home/osmc/.hts/tvheadend/passwd/* > /dev/null; then
+		echo Creating default password for user osmc
+		mkdir -p /home/osmc/.hts/tvheadend/passwd
+		cat <<- 'EOF' > /home/osmc/.hts/tvheadend/passwd/c247bc5760446f232f0d4853fda9c1ff
+		{
+			"enabled": true,
+			"username": "osmc",
+			"password2": "VFZIZWFkZW5kLUhpZGUtb3NtYw==",
+			"wizard": true
+		}
+
+		EOF
+	fi
+
+	# make sure ownerships are good
+	chown -R osmc:video /home/osmc/.hts/
+
+	if ischroot; then
```

Comment:
ischroot is used to detect whether we're installing this at (usually) build time. Doesn't make much sense here. OSMC doesn't currently (no plans to do so) bundle TVH by default, so this isn't needed.